lib/expline.ex

```diff
@@ -13,19 +13,25 @@ defmodule Expline do
   building, read the `Expline.Spline` module documentation.
   """
 
-  @typep state() :: Expline.Spline.t
+  @typep state() :: Expline.Spline.t()
 
   @doc """
   Builds a spline from the provided list of points and holds the state in a
   process without links (outside of a supervision tree).
 
   See `start_link/2` for more information.
   """
-  @spec start(list(Expline.Spline.point()), GenServer.options()) :: {:ok, pid()}
-                                                                  | {:error, {:already_started, pid()}}
-                                                                  | {:error, Expline.Spline.creation_error()}
+  @spec start(list(Expline.Spline.point()), {:graceful_shutdown, boolean()} | GenServer.options()) ::
+          {:ok, pid()}
+          | {:error, {:already_started, pid()}}
+          | {:error, Expline.Spline.creation_error()}
   def start(points, opts \\ []) do
-    GenServer.start(__MODULE__, [points], opts)
+    {graceful_shutdown, opts} = Keyword.pop(opts, :graceful_shutdown, false)
+
+    case GenServer.start(__MODULE__, {graceful_shutdown, [points]}, opts) do
+      {:error, {:shutdown, reason}} -> {:error, reason}
+      other -> other
+    end
```

Comment:
This is (and below for `start_link/2`) **the only significant change**, everything else is formatter.